share/spice/github/github.js

```diff
@@ -6,73 +6,96 @@
           return Spice.failed('github');
         }
 
-        var script = $('[src*="/js/spice/github/"]')[0],
-                    source = $(script).attr("src"),
-                    query = source.match(/github\/([^\/]+)/)[1];
-
-        if (/language:".*?"/.test(unescape(query))) {
-            var itemType = "Git Repos (" + unescape(query).match(/language:"(.*?)"/)[1] + ")";
-        } else {
-            var itemType = "Git Repos";
-        }
-
         var results = api_result.data.items;
 
         if (!results) {
             return Spice.failed('github');
         }
 
-        // TODO: temp size limit - relevancy block should handle this later
-        if (results.length > 30)
-            results = results.splice(0,30);
+        if (results.length)
+            results = results[0];
 
-        DDG.require("moment.js", function() {
+        var key = results.full_name.replace(/^.*\//, '');
+        results.title = key;
+        results.url = results.html_url;
+        results.imageTile = true;
+        
+        var fatheads;
+        $.getJSON(DDG.get_asset_path('github', 'fatheads.json'), function (data) { fatheads = data; });
```

Comment:
The callback function here needs to encompass all of the code below. `fatheads` will only be defined when that file is successfully loaded and on a slow connection for example, that could take 1+ seconds, but the code will keep executing and we'd have problems:

```js
$.getJSON(DDG.get_asset_path('github', 'fatheads.json'), function (data) { 
    var fatheads = data;
    DDG.require("moment.js", function() { ...
    });
});
```